core/operation/src/main/java/uk/gov/gchq/gaffer/operation/OperationChain.java

```diff
@@ -184,7 +184,11 @@ public Map<String, String> getOptions() {
 
     @Override
     public void setOptions(final Map<String, String> options) {
-        this.options = options;
+        if (options == null) {
+            this.options = new HashMap<>();
```

Comment:
Why do this if it is now set to a `HashMap` by default; is this in case it's set to something else already?

I'm guessing the new default `HashMap` is because of interaction with another class?

Reply:
Its mostly due to the proxy store as it was asking for the current options (which were null) then attempting to add to them which hit a null pointer. It's really its more a bug here as shouldn't be able to set the options to null they should just be blank/empty.